include/random_walks/gaussian_rdhr_walk.hpp

```diff
@@ -0,0 +1,117 @@
+// VolEsti (volume computation and sampling library)
+
+// Copyright (c) 2012-2020 Vissarion Fisikopoulos
+// Copyright (c) 2018-2020 Apostolos Chalkis
+
+// Contributed and/or modified by Apostolos Chalkis, as part of Google Summer of Code 2018 program.
+
+// Licensed under GNU LGPL.3, see LICENCE file
+
+#ifndef RANDOM_WALKS_GAUSSIAN_RDHR_WALK_HPP
+#define RANDOM_WALKS_GAUSSIAN_RDHR_WALK_HPP
+
+#include "generators/boost_random_number_generator.hpp"
+#include "random_walks/gaussian_helpers.hpp"
+
+// Pick a point from the distribution exp(-a_i||x||^2) on the chord
+template
+<
+    typename Point,
+    typename NT,
+    typename RandomNumberGenerator
+>
+void chord_random_point_generator_exp(Point &lower,
+                                      Point & upper,
+                                      const NT &a_i,
+                                      Point &p,
+                                      RandomNumberGenerator& rng)
+{
+    NT r, r_val, fn;
+    const NT tol = 0.00000001;
```

Comment:
Same comment as before. It is better to set tol outside the function so that all algorithms see the same value.